src/vivarium/framework/results/context.py

```diff
@@ -95,7 +95,7 @@ def add_observation(
         additional_stratifications: List[str],
         excluded_stratifications: List[str],
         when: str,
-        report: Callable[..., None],
+        report: Callable[[str, pd.DataFrame], None],
```

Comment:
Did we decide this should be `Callable[[str, pd.DataFrame], None]` and not `Callable[[str, Any], None]`?

Reply:
for now at least, ResultsContext.gather_results does indeed return specifically a dataframe